gpt-survey-summarizer/values.yaml

```diff
@@ -0,0 +1,108 @@
+# Default values for gpt-survey-summarizer.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+config:
+  # -- Openai API Key
+  openAiApiKey: "" 
+  discord:
+    # -- Discord Application ID
+    clientId: ""
+    # -- Discord Guild/Server ID
+    guildId: ""
+    # -- Discord API Token
+    token: ""
+
+redis:
+  # -- The redis architecture (accepted values are: standalone, replication)
+  architecture: standalone
+  auth:
+    # -- Redis password
+    password: ""
+  # -- Redis hostname
+  host: ""
+  # -- Redis port
+  port: 6379
```

Comment:
**suggestion:** I suggest to remove these values given they are not defined in the `redis` sub-chart. The `redis.` namespace is maintained by the `redis` chart developers, and we wouldn't want collide with them. (I can see one exception, being `redis.enabed`, when/if needed).

Also, we wouldn't want the `helmfile` to have to specify the `host` address.

I have put together a patch to declare an `redisUrl` helper, load it to the `Secret`, and inject the `redisUrl` as an env variable for the `Deployments`.

```diff
diff --git a/gpt-survey-summarizer/templates/_helpers.tpl b/gpt-survey-summarizer/templates/_helpers.tpl
index 02260232..870d25c3 100644
--- a/gpt-survey-summarizer/templates/_helpers.tpl
+++ b/gpt-survey-summarizer/templates/_helpers.tpl
@@ -60,3 +60,7 @@ Create the name of the service account to use
 {{- default "default" .Values.serviceAccount.name }}
 {{- end }}
 {{- end }}
+
+{{- define "redisUrl" -}}
+  redis://:{{ .Values.redis.auth.password }}@{{ include "gpt-survey-summarizer.fullname" . }}-redis-master:6379
+{{- end -}}
diff --git a/gpt-survey-summarizer/templates/deployment_bot.yaml b/gpt-survey-summarizer/templates/deployment_bot.yaml
index 24a391de..7479a990 100644
--- a/gpt-survey-summarizer/templates/deployment_bot.yaml
+++ b/gpt-survey-summarizer/templates/deployment_bot.yaml
@@ -50,7 +50,10 @@ spec:
                   name: {{ include "gpt-survey-summarizer.fullname" . }}
                   key: discordGuildId
             - name: REDIS_URL
-              value: redis://:{{ .Values.redis.auth.password }}@{{ .Values.redis.host }}:{{ .Values.redis.port }}
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "gpt-survey-summarizer.fullname" . }}
+                  key: redisUrl
           {{- if .Values.bot.extraEnvVars }}
             {{- toYaml .Values.bot.extraEnvVars | nindent 12 }}
           {{- end }}
diff --git a/gpt-survey-summarizer/templates/deployment_server.yaml b/gpt-survey-summarizer/templates/deployment_server.yaml
index f6a0a96c..c9c3ba20 100644
--- a/gpt-survey-summarizer/templates/deployment_server.yaml
+++ b/gpt-survey-summarizer/templates/deployment_server.yaml
@@ -40,7 +40,10 @@ spec:
                   name: {{ include "gpt-survey-summarizer.fullname" . }}
                   key: openAiApiKey
             - name: REDIS_URL
-              value: redis://:{{ .Values.redis.auth.password }}@{{ .Values.redis.host }}:{{ .Values.redis.port }}
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "gpt-survey-summarizer.fullname" . }}
+                  key: redisUrl
           {{- if .Values.server.extraEnvVars }}
             {{- toYaml .Values.server.extraEnvVars | nindent 12 }}
           {{- end }}
diff --git a/gpt-survey-summarizer/templates/secret.yaml b/gpt-survey-summarizer/templates/secret.yaml
index 50b90c0a..5beb24d6 100644
--- a/gpt-survey-summarizer/templates/secret.yaml
+++ b/gpt-survey-summarizer/templates/secret.yaml
@@ -7,3 +7,5 @@ data:
   discordClientId: {{ .Values.config.discord.clientId | b64enc }}
   discordGuildId: {{ .Values.config.discord.guildId | b64enc }}
   discordToken: {{ .Values.config.discord.token | b64enc }}
+  redisPassword: {{ .Values.redis.auth.password | b64enc }}
+  redisUrl: {{ include "redisUrl" . | b64enc }}
diff --git a/gpt-survey-summarizer/values.yaml b/gpt-survey-summarizer/values.yaml
index 1f2f1ad6..5115a519 100644
--- a/gpt-survey-summarizer/values.yaml
+++ b/gpt-survey-summarizer/values.yaml
@@ -19,10 +19,6 @@ redis:
   auth:
     # -- Redis password
     password: ""
-  # -- Redis hostname
-  host: ""
-  # -- Redis port
-  port: 6379
 
 image:
   # -- The repository of the image
```

Reply:
Applied.